### DISCUSSION TITLE
VSCode extension for power fx

### DISCUSSION THREAD

**Discussion by MikeStall** [microsoft/Power-Fx#227] [General]
A few folks reached out to me about a VS Code extension for Power Fx (https://powerusers.microsoft.com/t5/Power-Apps-Ideas/VSCode-extension-for-Power-Fx-formulas-syntax-highlighting/idi-p/1504407) 

It's a great idea and something on our roadmap.

There are 2 projects that really need to converge here. 
1. **We generate *.fx.yaml files** from a Canvas App (see https://github.com/microsoft/PowerApps-Language-Tooling). These contain the formulas and can be edited with any yaml editor, but that's still missing editor support for the fx expressions inside the yaml. 
The .fx.yaml extension was guidance from the VSCode team - it's still a yaml file, so it needs to have a .yaml extension, but the editor also needs to know that it's "fx enabled", hence then .fx.yaml. 

As Power Fx goes into other products, there may be other products that also export fx.yaml files. This is an area we need more clarity on. 

2. **There is a rich formula bar control for Power Fx and Language Service Protocol (LSP) support**.  We have a demo of it here: https://github.com/microsoft/power-fx-host-samples/tree/main/Samples/WebDemo . The LSP support is at https://github.com/microsoft/Power-Fx/tree/main/src/libraries/Microsoft.PowerFx.LanguageServerProtocol

Part of our motivation to implement the formula bar control using LSP was to enable a future VSCode extension.  We just haven't got there yet ... we first need need to get Power Fx to GA (out of the preview), 

All of this is open source so we do welcome community explorations here.  ;)

**Comment by lucahttp** [comment]
Very interesting

**Comment by timnudecision** [comment]
It would be useful to simply have an enhanced version of the WebDemo REPL running as a VSCode extension: Intellisense for editing Power-Fx with an execution terminal at the bottom.  That would make it much easier to code larger programs in PowerFx.

I am really looking forward to the GA release and the YAML extension for Canvas apps.

**Comment by ghost** [comment]
Are there any new developments going on? 
Came across this thread today and this feature/extension would be so handy!

**Comment by briankinsella** [comment]
+1 for adding this feature! Am surprised not already baked in. I'm debugging Power FX now in a sprawling canvas app in VS Code. Full Intellisense would be sweet. Related: not finding an fx.yaml JSON schema definition on GitHub to otherwise reference in VS Code - suppose that's the 1st step. Thanks for pursuing this idea!

**Comment by anderson-joyle** [comment]
Power Fx team is currently focused on releasing our 1.0 version (out of the preview) and also supporting products powered by Power Fx that are about to go generally available. Power Fx is an open-source project, and we do encourage contributions.

**Comment by snizar007** [comment]
Update: 
To use Power Fx in a command mode, one can use Power Platform extension in VSCode. Make sure you get PAC CLI version 1.28.x to be able to use PAC POWER-FX capabilities. Try it and share your thoughts at https://aka.ms/pac/powerfx

**Reply by K-Schumacher** [reply]
I believe there is a syntax error in question 5 "We would you like to collect scenarios you would like to see being enabled?" I'm assuming it's intended to read as "We would like to collect scenarios you would like to see being enabled." or "What are some scenarios you would like to see being enabled?" but please correct me if I am wrong.

**Comment by AdityaBansal1402** [comment]
So guys, What's been the update? Is powerfx vs code extension dead or what?? 💀
I  would be really interested in a roadmap that I can view if there is any 💯

**Reply by filcole** [reply]
The [Yaml Embedded Languages](https://marketplace.visualstudio.com/items?itemName=harrydowning.yaml-embedded-languages) VSCode extension contributed by @devkeydet is an improvement in this area.  See this PR: https://marketplace.visualstudio.com/items?itemName=harrydowning.yaml-embedded-languages

**Reply by lucahttp** [reply]
also you have [https://github.com/rajyraman/Power-Fx-.NET-Interactive](https://github.com/rajyraman/Power-Fx-.NET-Interactive) where you can run formulas inside vscode

**Comment by This-Is-NPC** [comment]
An MVP released this extension a few months ago

Extension ID: PowerAppsTools.powerfx

![image](https://github.com/user-attachments/assets/8a3d30b9-7492-4bb7-bf15-735eb70c8ece)
